apps/web/src/utils/firebase.ts

```diff
@@ -47,16 +53,24 @@ export const fireCache: FireCache = {
 	idToken: undefined,
 };
 
-auth.onIdTokenChanged(async (user) => {
-	if (user) {
-		fireCache.idToken = await user.getIdToken();
-	} else {
-		fireCache.idToken = undefined;
-	}
-});
-
-export const initializeAuth = async () => {
+export const initializeAuth = async (settings?: SettingDocument) => {
 	if (auth.currentUser) {
 		fireCache.idToken = await auth.currentUser.getIdToken();
 	}
+
+	if (settings?.profile.email) {
+		auth.onIdTokenChanged(async (user) => {
+			if (user) {
+				fireCache.idToken = await user.getIdToken();
+
+				if (appState.remoteConfig.deepAnalyticsEnabled) {
+					await universalActions.syncRemoteProfile();
+				}
+			} else {
+				fireCache.idToken = undefined;
+			}
+		});
+	}
```

Comment:
this auth event listener should not be initialized in `initializeAuth`, when users sign in, they need to reload the app to sync the analytics. Need to put it in global scope or module scope.